multinode_integration_tests/src/main.rs

```diff
@@ -342,6 +349,104 @@ impl MockNode {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+struct ProbeTarget {
+    ip_address: IpAddr,
+    port_specs: Vec<PortSpec>,
+}
+
+impl FromStr for ProbeTarget {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut main_pieces = s.split(':').collect_vec();
+        if main_pieces.len() != 2 {
+            return Err("Syntax: <IP address>:<U|T><port>/<U|T><port>/...".to_string());
+        }
+        let ip_address_string = main_pieces.remove(0);
+        let port_specs_string = main_pieces.remove(0);
+        let ip_address = match IpAddr::from_str(ip_address_string) {
+            Ok(ip_addr) => ip_addr,
+            Err(_) => return Err("Syntax: <IP address>:<U|T><port>/<U|T><port>/...".to_string()),
+        };
+        let port_spec_strings = port_specs_string.split('/').collect_vec();
+        let port_spec_results = port_spec_strings
+            .into_iter()
+            .map(PortSpec::from_str)
+            .collect_vec();
```

Comment:
Also I understand why using `.collect_vec()`  feels cool but has an disadvantage. It does not show you what type the collected items have. You have to read it out from elsewhere or simply guess. Here I caught myself wondering because of the next chain of iterators for finding the errors. Knowing the type would've made my happier.

Reply:
Okay, if you like...